tsinfer/formats.py

```diff
@@ -1185,9 +1203,13 @@ def from_tree_sequence(cls, ts, use_times=True, use_metadata=True, **kwargs):
                     samples_metadata=[sample_metadata],
                 )
         for v in ts.variants():
-            variant_time = None
```

Comment:
Can't this be done by just changing ``variant.time = USE_FREQ_AS_TIME`` on the first line? As in, there's no need to change the logic, we just change the marker value.

Reply:
I'm not sure what you mean here. I don't think a `ts.Variant` object has a `time` attribute?

Reply:
Ping @jeromekelleher in case this conversation is muted

Reply:
I meant ``variant_time``

Reply:
@jeromekelleher - Oh, I see. You mean the same as the others, take the stuff out of the `else` clause(s) as before. Something like:

```
        for v in ts.variants():
            variant_time = USE_FREQ_AS_TIME
            if use_times:
                variant_time = self.MEANINGLESS_TIME
                if len(v.site.mutations) == 1:
                    variant_time = ts.node(v.site.mutations[0].node).time
        ...
```

Reply:
Yes - see point below about taking this out of SampleData entirely though.